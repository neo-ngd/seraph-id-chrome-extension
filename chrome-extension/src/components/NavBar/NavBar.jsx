// Copyright (c) 2019 Swisscom Blockchain AG
// Licensed under MIT License

import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import { AppBar, Toolbar, Box, ButtonBase } from '@material-ui/core';
import image from '../../assets/icons/fingerprint.png';

/**
 * Component styles
 * @type {StylesHook<Styles<{readonly spacing?: *, readonly palette?: *}, {}, string>>}
 */
const useStyles = makeStyles(({ palette, spacing }) => ({
  barStyle: {
    background: palette.primary.dark,
    height: '56px',
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
    background: 'transparent',
    boxShadow: 'none',
  },
  imageContainer: {
    position: 'absolute',
    right: spacing(-2),
    top: spacing(1),
  },
  img: {
    maxWidth: '80px',
  },
}));

/**
 * <NavBar />
 * The navigation bar which shows the current account address and allows for opening the accounts dialog modal.
 * @param address
 * @param name
 * @param onOpenAccountsModal
 * @return {*}
 * @constructor
 */
const NavBar = ({ address, name, onOpenAccountsModal }) => {
  const classes = useStyles();

  return (
    <AppBar className={classes.barStyle} position="static" color="default">
      <Toolbar>
        <Box display="flex" flexDirection="column" flex={1}>
          <Box textAlign="center" color="text.primary" fontSize="13px">
            {name}
          </Box>

          <Box
            color="text.secondary"
            fontSize="12px"
            textAlign="center"
            pt={0.6}
          >
            {address}
          </Box>

          <ButtonBase
            disableRipple
            onClick={onOpenAccountsModal}
            className={classes.imageContainer}
          >
            <img alt="logo-img" className={classes.img} src={image} />
          </ButtonBase>
        </Box>
      </Toolbar>
    </AppBar>
  );
};

NavBar.propTypes = {
  address: PropTypes.string,
  name: PropTypes.string,
  onOpenAccountModal: PropTypes.func,
};

export default NavBar;
