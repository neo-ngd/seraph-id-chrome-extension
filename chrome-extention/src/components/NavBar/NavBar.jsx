import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import {AppBar, Toolbar, Box} from '@material-ui/core';

const useStyles = makeStyles(({ palette }) => ({
  barStyle: {
    background: palette.primary.dark,
    height: '56px',
    position: 'absolute',
    top: 0,
    left: 0,
    right: 0,
  },
}));

export default function NavBar({ address, name }) {
  const classes = useStyles();

  return (
    <AppBar className={classes.barStyle} position="static" color="default">
      <Toolbar>
        <Box display="flex" flexDirection="column" flex={1}>
          <Box textAlign="center" color="text.primary" fontSize="11px">
            {name}
          </Box>

          <Box color="text.secondary" fontSize="10px" textAlign="center" pt="5px">
            {address}
          </Box>
        </Box>
      </Toolbar>
    </AppBar>
  );
}
