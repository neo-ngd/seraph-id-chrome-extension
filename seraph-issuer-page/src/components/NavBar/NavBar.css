/* 
Copyright (c) 2019 Swisscom Blockchain AG
Licensed under MIT License
*/

.NavBarRoot {
  flex-grow: 1;
}

.NavBarTypography {
  flex-grow: 1;
}

.HelpButton {
  text-decoration: none;
  color: white;
}

.navLogo {
  height: 50px;
}

.NavBarMain {
  background-color: black !important;
}

.HelpIconBar {
  font-size: 33px !important;
}
