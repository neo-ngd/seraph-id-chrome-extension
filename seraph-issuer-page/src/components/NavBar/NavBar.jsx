import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import {AppBar, Toolbar, IconButton, Typography, Box} from "@material-ui/core";
import AccountBalanceIcon from "@material-ui/icons/AccountBalance";
import {Link} from "react-router-dom";
import Address from "../Adress/Address";

const useStyles = makeStyles(({ palette, spacing }) => ({
  barStyle: {
    background: palette.primary.dark,
    height: "56px",
    position: "absolute",
    top: 0,
    left: 0,
    right: 0,
  },
  toolbar: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  left: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',

  },
  links: {

  },
  link: {
    color: 'white',
    margin: spacing(1),
    '&:hover': {
      textDecoration: 'none',
    }
  }
}));

export default function NavBar({ address }) {
  const classes = useStyles();

  return (
    <AppBar className={classes.barStyle} position="static" color="default">
      <Toolbar className={classes.toolbar}>
        <Box className={classes.left}>
          <IconButton color="inherit" aria-label="Menu">
            <AccountBalanceIcon className="GovernmentLogo" />
          </IconButton>
          <Typography
            style={{ color: "#ffffffff" }}
            variant="h6"
            color="inherit"
            className="NavBarTypography"
          >
            {" "}
            Government Web Page{" "}
          </Typography>
        </Box>
        <Address address={address} />
        <Box className={classes.links}>
          <Link className={classes.link} to={'/government'}>issuer</Link>
          <Link className={classes.link} to={'/verifier'}>verifier</Link>
        </Box>
      </Toolbar>
    </AppBar>
  );
}
