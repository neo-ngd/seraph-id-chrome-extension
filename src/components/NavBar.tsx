import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Grid from '@material-ui/core/Grid';
import logo from "../assets/logo.png"

const useStyles = makeStyles({
  root: {
    flexGrow: 1,
  },
  barStyle:{
    background:"black"
  },
  imageStyle:{
    height:"7vh"
  }
});

export default function NavBar() {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <AppBar className={classes.barStyle} position="static" color="default">
        <Toolbar>
        <Grid container spacing={3}>
        <Grid item xs>
        </Grid>
        <Grid item xs>
        <img className={classes.imageStyle} src={logo} alt="Logo" />
        </Grid>
        <Grid item xs>
        </Grid>
      </Grid>
        </Toolbar>
      </AppBar>
    </div>
  );
}