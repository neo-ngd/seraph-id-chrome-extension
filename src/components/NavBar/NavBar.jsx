import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Grid from '@material-ui/core/Grid';

const useStyles = makeStyles({
  root: {
    flexGrow: 1,
  },
  barStyle: {
    background: 'black',
  },
});

export default function NavBar() {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <AppBar className={classes.barStyle} position="static" color="default">
        <Toolbar>
          <Grid container spacing={3}></Grid>
        </Toolbar>
      </AppBar>
    </div>
  );
}
