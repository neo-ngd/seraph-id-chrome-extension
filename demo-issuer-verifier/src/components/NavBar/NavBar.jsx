// Copyright (c) 2019 Swisscom Blockchain AG
// Licensed under MIT License

import React from "react";
import "./NavBar.css";
import { Link } from "react-router-dom";
import {
  Tooltip,
  AppBar,
  Toolbar,
  Typography,
  IconButton
} from "@material-ui/core";
import HelpIcon from "@material-ui/icons/HelpOutline";
import logoHorizontal from "../../assets/seraph-logo-horizontal.png";

function NavBar() {
  return (
    <div className="NavBarRoot">
      <AppBar position="static" className="NavBarMain">
        <Toolbar>
          <Tooltip title="Go to Seraph ID landing page">
            <a href="https://www.seraphid.io/">
              <img
                src={logoHorizontal}
                alt="SeraphID logo"
                className="navLogo"
              />
            </a>
          </Tooltip>
          <Typography className="NavBarTypography"> </Typography>
          <Tooltip title="Help">
            <Link to="/help" className="HelpButton">
              <IconButton color="inherit" aria-label="Menu">
                <HelpIcon className="HelpIconBar" />
              </IconButton>
            </Link>
          </Tooltip>
        </Toolbar>
      </AppBar>
    </div>
  );
}

export default NavBar;
