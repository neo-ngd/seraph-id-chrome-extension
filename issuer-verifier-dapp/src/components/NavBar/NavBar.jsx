// Copyright (c) 2019 Swisscom Blockchain AG
// Licensed under MIT License

import React from "react";
import "./NavBar.css";
import {
  Tooltip,
  AppBar,
  Toolbar,
  Typography,
  IconButton
} from "@material-ui/core";
import HelpIcon from "@material-ui/icons/HelpOutline";
import logoHorizontal from "../../assets/seraph-logo-horizontal.png";

function NavBar() {
  return (
    <div className="NavBarRoot">
      <AppBar position="static" className="NavBarMain">
        <Toolbar>
          <Tooltip title="Go to Seraph ID landing page">
            <a href="https://www.seraphid.io/">
              <img
                src={logoHorizontal}
                alt="SeraphID logo"
                className="navLogo"
              />
            </a>
          </Tooltip>
          <Typography className="NavBarTypography"> </Typography>
          <Tooltip title="Help">
            <IconButton color="inherit" aria-label="Menu">
              <HelpIcon className="HelpIconBar" />
            </IconButton>
          </Tooltip>
        </Toolbar>
      </AppBar>
    </div>
  );
}

export default NavBar;
