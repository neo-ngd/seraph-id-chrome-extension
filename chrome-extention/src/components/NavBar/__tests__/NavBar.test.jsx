import React from 'react';
import { shallow } from 'enzyme';
import NavBar from "../NavBar";

const props = {
    address: 'test',
    name: 'test',
    onOpenAccountModal: jest.fn(),
};

describe('<NavBar />', () => {
    it('should render component', () => {
        const wrapper = shallow(<NavBar {...props} />);
        expect(wrapper).toMatchSnapshot()
    });
});
