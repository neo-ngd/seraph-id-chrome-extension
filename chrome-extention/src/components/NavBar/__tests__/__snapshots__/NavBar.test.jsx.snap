// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> should render component 1`] = `
<WithStyles(ForwardRef(AppBar))
  className="makeStyles-barStyle-1"
  color="default"
  position="static"
>
  <WithStyles(ForwardRef(Toolbar))>
    <Styled(MuiBox)
      display="flex"
      flex={1}
      flexDirection="column"
    >
      <Styled(MuiBox)
        color="text.primary"
        fontSize="11px"
        textAlign="center"
      >
        test
      </Styled(MuiBox)>
      <Styled(MuiBox)
        color="text.secondary"
        fontSize="10px"
        pt={0.6}
        textAlign="center"
      >
        test
      </Styled(MuiBox)>
      <WithStyles(ForwardRef(ButtonBase))
        className="makeStyles-imageContainer-2"
        disableRipple={true}
      >
        <img
          alt="logo-img"
          className="makeStyles-img-3"
          src="test-file-stub"
        />
      </WithStyles(ForwardRef(ButtonBase))>
    </Styled(MuiBox)>
  </WithStyles(ForwardRef(Toolbar))>
</WithStyles(ForwardRef(AppBar))>
`;
